src/StoreBase.ts

```diff
@@ -60,178 +56,211 @@ export abstract class StoreBase {
-                const existingKeys = this._gatheredCallbacks.get(callback);
-                if (existingKeys === undefined) {
-                    this._gatheredCallbacks.set(callback, _.clone(keys!!!));
-                } else if (existingKeys === null) {
+                const existingMeta = StoreBase._pendingCallbacks.get(callback);
+                StoreBase._updateExistingMeta(existingMeta, throttledUntil, bypassBlock);
+
+                if (existingMeta === undefined) {
+                    StoreBase._pendingCallbacks.set(callback, { keys: _.clone(keys), throttledUntil, bypassBlock });
+                } else if (existingMeta.keys === null) {
                     // Do nothing since it's already an all-key-trigger
                 } else {
                     // Add them all to the end of the list
                     _.forEach(keys, key => {
-                        existingKeys.push(key);
+                        existingMeta.keys!!!.push(key);
                     });
                 }
             });
 
             _.forEach(this._autoSubscriptions[StoreBase.Key_All], sub => {
-                const existingKeys = this._gatheredCallbacks.get(sub.callback);
-                if (existingKeys === undefined) {
-                    this._gatheredCallbacks.set(sub.callback, _.clone(keys!!!));
-                } else if (existingKeys === null) {
+                const existingMeta = StoreBase._pendingCallbacks.get(sub.callback);
+                StoreBase._updateExistingMeta(existingMeta, throttledUntil, bypassBlock);
+
+                if (existingMeta === undefined) {
+                    StoreBase._pendingCallbacks.set(sub.callback, { keys: _.clone(keys), throttledUntil, bypassBlock });
+                } else if (existingMeta.keys === null) {
                     // Do nothing since it's already an all-key-trigger
                 } else {
                     // Add them all to the end of the list
                     _.forEach(keys, key => {
-                        existingKeys.push(key);
+                        existingMeta.keys!!!.push(key);
                     });
                 }
             });
         }
 
-        if (this._throttleMs && !StoreBase._bypassThrottle) {
-            // Needs to accumulate and trigger later -- start a timer if we don't have one running already
-            // If there are no callbacks, don't bother setting up the timer
-            if (!this._throttleTimerId && this._gatheredCallbacks.size !== 0) {
-                this._throttleTimerId = Options.setTimeout(this._resolveThrottledCallbacks, this._throttleMs);
-            }
-        } else {
-            // No throttle timeout, so just resolve now
-            this._resolveThrottledCallbacks();
+        if (!throttledUntil || this._bypassTriggerBlocks) {
+            StoreBase._resolveCallbacks();
         }
     }
 
     private _resolveThrottledCallbacks = () => {
```

Comment:
Consider renaming `_resolveThrottledCallbacks` to avoid confusion with `_resolveCallbacks`. No one should call this other than the `setTimeout`, so perhaps `_handleThrottleTimeout` or something.